### PR TITLE
Make hypershift jobs match job naming guidelines

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-886](https://issues.redhat.com//browse/TRT-886)

This makes the hypershift jobs match more closely our naming guidelines.

- Deduplicate name as much as possible
- Move 4.13 and 4.14 from main to release-X.Y branches
- Remove release name from variant
- Consistent field ordering (`(top-level-test-suite)-(platform)-(cni)-(extras)`)

See https://docs.ci.openshift.org/docs/how-tos/naming-your-ci-jobs/ for more info

Example name change that happens here:

Old name: periodic-ci-openshift-hypershift-main-periodics-4.13-conformance-aws-ovn
New name: periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-aws-ovn-conformance

Old name: periodic-ci-openshift-hypershift-main-periodics-4.13-e2e-aws-periodic
New name: periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-aws-ovn

### CODE REVIEW

**Comment by sjenning** [comment]
wow, I think we may have done mostly the same thing https://github.com/openshift/release/pull/37585

File: ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml [hunk 1]
```diff
@@ -1,1364 +0,0 @@
...
-  - base_ref: main
-    org: openshift
-    repo: hypershift
-  labels:
-    ci-operator.openshift.io/variant: periodics-4.13
-    ci.openshift.io/generator: prowgen
-    job-release: "4.13"
-    pj-rehearse.openshift.io/can-be-rehearsed: "true"
-  name: periodic-ci-openshift-hypershift-main-periodics-4.13-conformance-aws-ovn
-  reporter_config:
```
Comment (sjenning) [inline ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml:100]: we manually added these `reporter_config` blocks to the `jobs` file and they didn't care over

Comment (stbenjam) [reply ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml:100]: Thanks, I'll add these back

**Comment by smg247** [comment]
@stbenjam, this seems to be making the `periodic-prow-auto-aggregator-job-names` job fail:
```
time="2023-03-27T18:00:34.200Z" level=fatal msg="Command failed" error="error reading https://raw.githubusercontent.com/openshift/release/master/ci-operator/jobs/openshift/hypershift/openshift-hypershift-main-periodics.yaml: 404"
``` See [logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/periodic-prow-auto-aggregator-job-names/1640413439341891584).

**Pull request merged**